lib/testing/gpuSimulateN2kPL1bitCorr.cpp

```diff
@@ -0,0 +1,212 @@
+                    for (int j_out = 0; j_out <= i_out; j_out++) {
+                        for (int i_in = 0; i_in < _blocksize; i_in++) {
+                            for (int j_in = 0; j_in < _blocksize; j_in++) {
+                                int i = i_out * _blocksize + i_in;
+                                int j = j_out * _blocksize + j_in;
+
+                                int32_t c = 0;
+                                for (int t64_in = 0; t64_in < nt64_inner; t64_in++) {
+                                    int t64 = t_out * nt64_inner + t64_in;
+                                    int t_pl = t64;
+                                    int t_rfi_lo = t64 & 0xF;
+                                    int t_rfi_hi = t64 >> 4;
+
+                                    uint64_t pl_i = pl_mask[t_pl * dt_pl + f * df_pl + i];
+                                    uint64_t pl_j = pl_mask[t_pl * dt_pl + f * df_pl + j];
+                                    uint64_t rfi =
+                                        rfi_mask[t_rfi_hi * dt_rfi + f * df_rfi + t_rfi_lo];
+                                    uint64_t v = pl_i & pl_j & rfi;
+                                    std::bitset<64> v_bits(v);
+                                    c += v_bits.count();
+                                } // t64_in
+
+                                counts[t_out * dt_c + f * df_c + block_idx * db_c + i_in * diin_c
+                                       + j_in] = c;
+                            } // j_in
+                        } // i_in
+
+                        block_idx++;
+                    } // j_out
+                } // i_out
+
+                DEBUG("Done t_outer {:d} of {:d} (freq {:d} of {:d}, nt_inner={:d})...", t_out,
+                      n_integrations, f, nf, _sub_integration_ntime);
+            } // f
+        } // t_out
+
+
+        // Fetch input metadata
+        const std::shared_ptr<const metadataObject> mc_in =
+            input_plmask_buf->get_metadata(input_pl_frame_id);
+        const std::shared_ptr<const chordMetadata> meta_in =
+            (mc_in && metadata_is_chord(mc_in)) ? get_chord_metadata(mc_in) : nullptr;
+
+        // Create output metadata
+        output_buf->allocate_new_metadata_object(output_frame_id);
+        const std::shared_ptr<metadataObject> mc = output_buf->get_metadata(output_frame_id);
+        if (!mc) {
+            FATAL_ERROR("Buffer {:s} frame {:d} cannot allocate metadata", output_buf->buffer_name,
+                        output_frame_id);
+        }
+        assert(mc);
+        if (!metadata_is_chord(mc)) {
+            FATAL_ERROR("Buffer {:s} frame {:d} does not have CHORD metadata",
+                        output_buf->buffer_name, output_frame_id);
+        }
+        assert(metadata_is_chord(mc));
+        const std::shared_ptr<chordMetadata> meta_out = get_chord_metadata(mc);
+        assert(meta_out);
+
+        meta_out->set_name("cpusim_n2_counts");
```

Comment:
The `name` describes the physical quantity, not how it was calculated. I call it `n2k_counts`. `n2_counts` would also work. We should have a common set of names across all kernels.